fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -266,13 +266,20 @@ public ShowResultSet execute() throws AnalysisException {
     }
 
     private void handleShowMaterializedView() throws AnalysisException {
-        String dbName = ((ShowMaterializedViewStmt) stmt).getDb();
+        ShowMaterializedViewStmt showMaterializedViewStmt = (ShowMaterializedViewStmt) stmt;
+        String dbName = showMaterializedViewStmt.getDb();
         List<List<String>> rowSets = Lists.newArrayList();
 
         Database db = GlobalStateMgr.getCurrentState().getDb(dbName);
         if (db == null) {
             ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_DB_ERROR, dbName);
         }
+        if (!GlobalStateMgr.getCurrentState().getAuth().checkDbPriv(ctx, db.getFullName(), PrivPredicate.SHOW)) {
```

Comment:
better to check privilege in ShowStmtAnalyzer

Reply:
move code to PrivilegeChecker